File: src/mouse/entities.go
package main

import "math/rand"

const (
	EmojiSize        = 24.0
	CatSpeed         = 2.0
	MouseSpeed       = 5.0
	AirplaneSpeed    = 10.0
	HouseSize        = 57.0
	CherryDuration   = 5000 // Milliseconds
	AirplaneDuration = 5000 // Milliseconds
)

type Entity struct {
	X, Y      float64
	Speed     float64
	Emoji     string
	Active    bool
	Timer     int
	Direction struct {
		X, Y float64
	}
	Angle      float64
	Circling   bool
	SpawnCount int
	Duration   int
	Size       float64
}

func NewMouse(canvasWidth, canvasHeight int) Entity {
	return Entity{
		X:      float64(canvasWidth) / 2,
		Y:      float64(canvasHeight) / 2,
		Speed:  MouseSpeed,
		Emoji:  "🐭",
		Active: true,
	}
}

func NewCat(canvasWidth, canvasHeight int) Entity {
	return Entity{
		X:      rand.Float64()*(float64(canvasWidth)-40) + 20,
		Y:      rand.Float64()*(float64(canvasHeight)-40) + 20,
		Speed:  CatSpeed,
		Emoji:  "🐱",
		Active: false,
	}
}

func NewHouse(canvasWidth, canvasHeight int) Entity {
	return Entity{
		X:     float64(canvasWidth) / 2,
		Y:     float64(canvasHeight) / 2,
		Size:  HouseSize,
		Emoji: "🏠",
	}
}

func NewCheese(canvasWidth, canvasHeight int) Entity {
	return Entity{
		X:     rand.Float64()*(float64(canvasWidth)-40) + 20,
		Y:     rand.Float64()*(float64(canvasHeight)-40) + 20,
		Emoji: "🧀",
	}
}

func NewCherry() Entity {
	return Entity{
		Emoji:    "🍒",
		Active:   false,
		Duration: CherryDuration,
	}
}

func NewAirplane(canvasWidth, canvasHeight int) Entity {
	return Entity{
		X:        float64(canvasWidth) / 2,
		Y:        float64(canvasHeight) / 2,
		Speed:    AirplaneSpeed,
		Duration: AirplaneDuration,
		Emoji:    "✈️",
		Active:   false,
	}
}

File: static/entities.js
const CAT_SPEED = 2;
const MOUSE_SPEED = 5;
const AIRPLANE_SPEED = 10;
const HOUSE_SIZE = 57;
const CHERRY_DURATION = 5000;
const AIRPLANE_DURATION = 5000;


const mouse = {
    x: canvas.width / 2,
    y: canvas.height / 2,
    speed: MOUSE_SPEED,
    emoji: '🐭'
};

const cat = {
    x: Math.random() * (canvas.width - 40) + 20,
    y: Math.random() * (canvas.height - 40) + 20,
    speed: CAT_SPEED,
    active: false,
    timer: 0,
    direction: { x: 0, y: 0 },
    angle: 0,
    circling: false,
    emoji: '🐱'
};

const cheese = {
    x: Math.random() * (canvas.width - 40) + 20,
    y: Math.random() * (canvas.height - 40) + 20,
    emoji: '🧀'
};

const cherry = {
    x: null,
    y: null,
    active: false,
    spawnCounter: 0,            // Counts how many cheeses have been collected since last cherry spawn
    nextSpawn: 0,               // Number of cheeses needed until next cherry spawn
    duration: CHERRY_DURATION,   
    emoji: '🍒'
};

const house = {
    x: canvas.width / 2,
    y: canvas.height / 2,
    size: HOUSE_SIZE,
    emoji: '🏠'
};

const airplane = {
    x: canvas.width / 2,
    y: canvas.height / 2,
    active: false,
    startTime: null,
    speed: AIRPLANE_SPEED,       
    duration: AIRPLANE_DURATION,
    emoji: '✈️'
};

const airplaneIcon = {
    x: null,
    y: null,
    active: false,
    emoji: '✈️'
};